meteor/client/ui/Settings/components/FilterEditor.tsx

```diff
@@ -926,119 +796,898 @@ export default withTranslation()(
+							{t('Height')}
+							<EditAttribute
+								modifiedClassName="bghl"
+								attribute={`filters.${index}.height`}
+								obj={item}
+								type="float"
+								collection={RundownLayouts}
+								className="input text-input input-l"
+							/>
+						</label>
+					</div>
+					{scalable && (
+						<div className="mod mvs mhs">
+							<label className="field">
+								{t('Scale')}
+								<EditAttribute
+									modifiedClassName="bghl"
+									attribute={`filters.${index}.scale`}
+									obj={item}
+									type="float"
+									collection={RundownLayouts}
+									className="input text-input input-l"
+								/>
+							</label>
+						</div>
+					)}
+					<div className="mod mvs mhs">
+						<label className="field">
+							{t('Custom Classes')}
+							<EditAttribute
+								modifiedClassName="bghl"
+								attribute={`filters.${index}.customClasses`}
+								obj={item}
+								type="text"
+								collection={RundownLayouts}
+								className="input text-input input-l"
+								mutateDisplayValue={(v: string[] | undefined) => v?.join(',')}
+								mutateUpdateValue={(v: string | undefined) => v?.split(',')}
+							/>
+							<span className="text-s dimmed">
+								Add custom css classes for customization. Separate classes with a &lsquo;,&rsquo;
+							</span>
+						</label>
+					</div>
+				</React.Fragment>
+			)
+		}
+
+		render() {
+			const { t } = this.props
+
+			const isRundownLayout = RundownLayoutsAPI.isRundownLayout(this.props.item)
+			const isDashboardLayout = RundownLayoutsAPI.isDashboardLayout(this.props.item)
+
+			return (
+				<div className="rundown-layout-editor-filter mod pan mas" key={this.props.filter._id}>
+					<button
+						className="action-btn right mod man pas"
+						onClick={() => this.onRemoveElement(this.props.item, this.props.filter)}
+					>
+						<FontAwesomeIcon icon={faTrash} />
+					</button>
+					{isRundownLayout && (
+						<button
+							className={ClassNames('action-btn right mod man pas', {
+								star: (this.props.filter as any).default,
+							})}
+							onClick={() =>
+								this.onToggleDefault(
+									this.props.item as RundownLayout,
+									this.props.index,
+									!(this.props.filter as any).default
+								)
+							}
+						>
+							<FontAwesomeIcon icon={faStar} />
+						</button>
+					)}
+					<div>
+						<div className="mod mvs mhs">
+							<label className="field">
+								{t('Type')}
+								<EditAttribute
+									modifiedClassName="bghl"
+									attribute={`filters.${this.props.index}.type`}
+									obj={this.props.item}
+									options={this.props.supportedFilters}
+									type="dropdown"
+									mutateDisplayValue={(v) => (v === undefined ? this.props.supportedFilters[0] : v)}
+									collection={RundownLayouts}
+									className="input text-input input-l"
+								></EditAttribute>
+							</label>
+						</div>
+					</div>
+					{RundownLayoutsAPI.isFilter(this.props.filter)
```

Comment:
I'm a fan of ternary boolean operator, but this is way too much.

Reply:
I've made an attempt to clean this up, it might be a bit more readable https://github.com/tv2/tv-automation-server-core/commit/752c0f904dcafb1bd078231122da6fc1bd0feedb